src/LAMMPS.jl

```diff
@@ -361,19 +361,149 @@ function extract_variable(lmp::LMP, name::String, group=nothing)
+
+Scatter the named per-atom, per-atom fix, per-atom compute, or fix property/atom-based entity in data to all processes.
+By default (when `ids=nothing`), this method scatters data to all atoms in consecutive order according to their IDs.
+The optional parameter `ids` determines to which subset of atoms the data will be scattered.
+
+Compute entities have the prefix `c_`, fix entities use the prefix `f_`, and per-atom entites have no prefix.
+
+!!! warning "Type Verification"
+    Due to how the underlying C-API works, it's not possible to verify the element data-type of fix or compute style data.
+    Supplying the wrong data-type will not throw an error but will result in nonsensical date being supplied to the LAMMPS instance.
+
+!!! warning "ids"
+    The optional parameter `ids` only works, if there is a map defined. For example by doing:
+    `command(lmp, "atom_modify map yes")`
+    However, LAMMPS only issues a warning if that's the case, which unfortuately cannot be detected through the underlying API.
+    Starting form LAMMPS version `17 Apr 2024` this should no longer be an issue, as LAMMPS then throws an error instead of a warning.
+"""
+function scatter!(lmp::LMP, name::String, data::VecOrMat{T}, ids::Union{Nothing, Array{Int32}}=nothing) where T<:Union{Int32, Float64}
+    @assert name != "mass" "scattering/gathering mass is currently not supported! Use `extract_atom()` instead."
+
+    count = _get_count(lmp, name)
+    _T = _get_T(lmp, name)
+
+    @assert ismissing(_T) || _T == T "Expected data type $_T got $T instead."
+
+    dtype = (T === Float64)
+    natoms = get_natoms(lmp)
+    ndata = isnothing(ids) ? natoms : length(ids)
+
+    if data isa Vector
+        @assert count == 1
+        @assert ndata == lenght(data)
+    else
+        @assert count == size(data,1)
+        @assert ndata == size(data,2)
+    end
+
+    if isnothing(ids)
+        API.lammps_scatter(lmp, name, dtype, count, data)
+    else
+        @assert all(1 <= id <= natoms for id in ids)
+        API.lammps_scatter_subset(lmp, name, dtype, count, ndata, ids, data)
+    end
+
+    check(lmp)
+end
+
+function _get_count(lmp::LMP, name::String)
+    # values taken from: https://docs.lammps.org/Classes_atom.html#_CPPv4N9LAMMPS_NS4Atom7extractEPKc
+
+    if startswith(name, r"[f,c]_")
+        if name[1] == 'c'
+            API.lammps_has_id(lmp, "compute", name[3:end]) != 1 && error("Unknown per atom compute $name")
+
+            count_ptr = API.lammps_extract_compute(lmp::LMP, name[3:end], API.LMP_STYLE_ATOM, API.LMP_SIZE_COLS)
+        else
+            API.lammps_has_id(lmp, "fix", name[3:end]) != 1 && error("Unknown per atom fix $name")
+
+            count_ptr = API.lammps_extract_fix(lmp::LMP, name[3:end], API.LMP_STYLE_ATOM, API.LMP_SIZE_COLS, 0, 0)
+        end
+        check(lmp)
+
+        count_ptr = reinterpret(Ptr{Cint}, count_ptr)
+        count = unsafe_load(count_ptr)
+    
+        return count == 0 ? 1 : count
```

Comment:
This is counter-intuitive.

Reply:
I used to have a comment there explaining why I do that. It must have taken it out accidentally at some point.
Basically a count of 0 means that the entity is a vector type instead of an array type. But because I want to preserve type stability I just treat it as a 1xN matrix instead. Hence returning 1 if count is 0.